### PR TITLE
feat(inputs.internet_speed): support packet loss

### PR DESCRIPTION
**Pull request by r3inbowari**
## Summary
1. Bump to new version to improve test precision and fix bugs.
2. Support packet_loss field.

## Checklist

- [x] No AI generated code was used in this PR

### CODE REVIEW

File: plugins/inputs/internet_speed/README.md [hunk 1]
```diff
@@ -63,24 +64,25 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
...
-| Latency        | latency    | float64 | ms   |
-| Jitter         | jitter     | float64 | ms   |
-| Location       | location   | string  | -    |
+| Name           | field name  | type    | Unit |
+|----------------|-------------|---------|------|
+| Download Speed | download    | float64 | Mbps |
+| Upload Speed   | upload      | float64 | Mbps |
+| Latency        | latency     | float64 | ms   |
+| Jitter         | jitter      | float64 | ms   |
+| Packet Loss    | packet_loss | float64 | pr   |
```
Comment (powersj) [inline plugins/inputs/internet_speed/README.md:73]: pr is percent?

**Review by powersj — COMMENTED** [review]
hi @r3inbowari thanks for the PR and updates, one question from my side in the docs, otherwise looks good. Thanks again!

File: plugins/inputs/internet_speed/README.md (hunk 1 above)
Comment (r3inbowari) [reply plugins/inputs/internet_speed/README.md:73]: In general, it has no unit because it is probability.

Comment (powersj) [reply plugins/inputs/internet_speed/README.md:73]: The closer to 1 means the greater the chance you had packet loss and if it is exactly 1 then you had packet loss?

Comment (r3inbowari) [reply plugins/inputs/internet_speed/README.md:73]: > The closer to 1 means the greater the chance you had packet loss and if it is exactly 1 then you had packet loss?

1 means all packets loss.

Comment (r3inbowari) [reply plugins/inputs/internet_speed/README.md:73]: Maybe we can record the details of packet loss into the fields.

Contains three additional fields：
1. total number of packets received by the remote.
2. dup packet number.
3. the maximum index value received by the remote.

Comment (powersj) [reply plugins/inputs/internet_speed/README.md:73]: > 1 means all packets loss.

That sounds like a percentage then. I'm fine with reporting a single field as a percentage, I just wanted to clarify the meaning of the values. The end user probably won't care as much about how many packets were sent.

**Review by powersj — APPROVED** [review]

File: plugins/inputs/internet_speed/README.md (hunk 1 above)
Comment (r3inbowari) [reply plugins/inputs/internet_speed/README.md]: percentage = probability * 100%, so it's a probability :)

Comment (r3inbowari) [reply plugins/inputs/internet_speed/README.md]: changed

Comment (powersj) [reply plugins/inputs/internet_speed/README.md]: Hmm I really don't like the term probability when you are reporting a hard-value that was observed once.

You are calculating that value by taking the total number of packets received and dividing it by total sent right? Why not multiple it by 100 and make it a percentage? If you look at ping output you get a % of packet loss and it seems that is what you are trying to show here.

Comment (r3inbowari) [reply plugins/inputs/internet_speed/README.md]: okay, i changed it.

Comment (powersj) [reply plugins/inputs/internet_speed/README.md]: Thanks!

**Review by powersj — APPROVED** [review]

File: plugins/inputs/internet_speed/internet_speed.go [hunk 2]
```diff
@@ -92,6 +100,7 @@ func (is *InternetSpeed) Gather(acc telegraf.Accumulator) error {
 		if err != nil {
 			return fmt.Errorf("upload test failed failed: %w", err)
 		}
+		pLoss, _ = analyzer.RunMulti(is.servers.Hosts())
```
Comment (srebhan) [inline plugins/inputs/internet_speed/internet_speed.go:103]: Can you please add a comment that the error will always be `nil` or check the error?

File: plugins/inputs/internet_speed/internet_speed.go [hunk 3]
```diff
@@ -101,14 +110,27 @@ func (is *InternetSpeed) Gather(acc telegraf.Accumulator) error {
 		if err != nil {
 			return fmt.Errorf("upload test failed failed: %w", err)
 		}
+		_ = analyzer.Run(is.server.Host, func(packetLoss *transport.PLoss) {
```
Comment (srebhan) [inline plugins/inputs/internet_speed/internet_speed.go:113]: Same here, please either add a comment that the error is always `nil` _or_, even better, check the error as this is future proof...

**Review by srebhan — COMMENTED** [review]
@r3inbowari thanks for your contribution! I do have some comments in the code, mostly on error checking...

File: plugins/inputs/internet_speed/internet_speed.go (hunk 2 above)
Comment (r3inbowari) [reply plugins/inputs/internet_speed/internet_speed.go]: ok, warning packet loss not applicable here.

**Review by srebhan — APPROVED** [review]
Thanks @r3inbowari!

**Comment by srebhan** [comment]
@powersj assigning it back to you for checking the code change...

**Pull request merged**